builtin/providers/aws/resource_aws_spot_instance_request_test.go

```diff
@@ -138,10 +178,26 @@ func testAccCheckAWSSpotInstanceRequestAttributes(
 	}
 }
 
+func testAccCheckAWSSpotInstanceRequestAttributesVPC(
+	sir *ec2.SpotInstanceRequest) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		if sir.LaunchSpecification.SubnetID == nil {
+			return fmt.Errorf("SubnetID was not passed, but should have been for this instance to belong to a VPC")
+		}
+		return nil
+	}
+}
+
 const testAccAWSSpotInstanceRequestConfig = `
+resource "aws_key_pair" "debugging" {
+	key_name = "tmp-key"
+	public_key = "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQD3F6tyPEFEzV0LX3X8BsXdMsQz1x2cEikKDEY0aIj41qgxMCP/iteneqXSIFZBp5vizPvaoIR3Um9xK7PGoW8giupGn+EPuxIA4cDM4vzOqOkiMPhz5XK0whEjkVzTo4+S0puvDZuwIsdiW9mxhJc7tgBNL0cYlWSYVkz4G/fslNfRPW5mYAM49f4fhtxPb5ok4Q2Lg9dPKVHO/Bgeu5woMc7RY0p1ej6D4CKFE6lymSDJpW0YHX/wqE9+cfEauh7xZcG0q9t2ta6F6fmX0agvpFyZo8aFbXeUBr7osSCJNgvavWbM/06niWrOvYX2xwWdhXmXSrbX8ZbabVohBK41 phodgson@thoughtworks.com"
+}
+
 resource "aws_spot_instance_request" "foo" {
 	ami = "ami-4fccb37f"
 	instance_type = "m1.small"
+	key_name = "${aws_key_pair.debugging.key_name}"
```

Comment:
I changed the default spot test to validate that passing a `key_name` works; since it doesn't seem to make sense to start an instance without one.